File: testdata/rules/MCPG002/vulnerable/web.py
import httpx
from mcp.server.fastmcp import FastMCP

mcp = FastMCP("web")


@mcp.tool()
async def fetch_page(url: str) -> str:
    """Fetch a web page and return its text."""
    async with httpx.AsyncClient() as client:
        resp = await client.get(url)
    return resp.text

File: testdata/rules/MCPG002/vulnerable/web.ts
import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import { z } from "zod";

const server = new McpServer({ name: "web", version: "1.0.0" });

server.registerTool(
  "fetch_url",
  { description: "Fetch a URL and return the body", inputSchema: { url: z.string().url() } },
  async ({ url }) => {
    const res = await fetch(url);
    const body = await res.text();
    return { content: [{ type: "text", text: body }] };
  }
);
